src/notation/internal/notationinteraction.cpp

```diff
@@ -689,24 +689,70 @@ void NotationInteraction::moveChordNoteSelection(MoveDirection d)
         return;
     }
 
-    EngravingItem* current = selection()->element();
-    if (!current || !(current->isNote() || current->isRest())) {
-        return;
-    }
+    auto removeDuplicates = [](std::vector<Element*>&elements) {
+        std::sort(elements.begin(), elements.end());
+        auto it = std::unique(elements.begin(), elements.end());
+        elements.erase(it, elements.end());
+    };
 
-    EngravingItem* chordElem;
-    if (d == MoveDirection::Up) {
-        chordElem = score()->upAlt(current);
-    } else {
-        chordElem = score()->downAlt(current);
+    auto& score = *score();
+    auto& selection = score.selection();
+    EngravingItem* currentSingle = selection.element();
+    EngravingItem* oldSingle = currentSingle;
+    bool isRange = selection.isRange();
+    std::vector<Element*> notes;
+
+    // Single traverse:
+    if (currentSingle && (currentSingle->isNote() || currentSingle->isRest())) {
+        EngravingItem* newSingle = score.moveAlt(currentSingle, d);
+        if (newSingle == currentSingle)
+            return;
+        while (newSingle && newSingle->isRest() && toRest(newSingle)->isGap()) {
+            newSingle = score.moveAlt(newSingle, d);
+            if (newSingle == old)
```

Comment:
Shouldn#t this be
```cpp
            if (newSingle == currentSingle)
```

Reply:
Oh, I think I meant oldSingle

Reply:
it looks like it doesn't matter. Looks like currentSingle wasn't changed. So yeah currentSingle